packages/lab/src/Flow/hooks/useNode.tsx

```diff
@@ -0,0 +1,191 @@
+    color: colorProp,
+    subtitle: subtitleProp,
+    nodeActions: nodeActionsProp,
+    inputs: inputsProp,
+    outputs: outputsProp,
+    groupId,
+    labels: labelsProps,
+    nodeToolbarProps,
+  } = props;
+
+  const { registerNode, unregisterNode } = useNodeMetaRegistry();
+
+  const labels = useLabels(DEFAULT_LABELS, labelsProps);
+
+  const inputs = useMemo(() => identifyHandles(inputsProp), [inputsProp]);
+
+  const outputs = useMemo(() => identifyHandles(outputsProp), [outputsProp]);
+
+  const { nodeGroups } = useFlowContext();
+
+  const node = useFlowNode();
+
+  const reactFlowInstance = useFlowInstance();
+
+  const nodeGroup = (groupId && nodeGroups && nodeGroups[groupId]) || undefined;
+
+  const title = titleProp || nodeGroup?.label;
+  const icon = iconProp || nodeGroup?.icon;
+  const color = getColor(colorProp || nodeGroup?.color);
+  const iconColor = isValidElement(icon)
+    ? getColor(icon.props.color || "base_dark")
+    : getColor("base_dark");
+  const subtitle = subtitleProp || node?.data.nodeLabel;
+
+  const [showActions, setShowActions] = useState(false);
+
+  const toggleShowActions = useCallback(() => {
+    setShowActions(!showActions);
+  }, [showActions]);
+
+  const getNodeToolbarProps = useCallback(
+    () => ({
+      offset: 0,
+      isVisible: showActions,
+      ...nodeToolbarProps,
+    }),
+    [nodeToolbarProps, showActions],
+  );
+
+  const nodeActions = useMemo(
+    () =>
+      nodeActionsProp || [
+        { id: "delete", label: labels?.deleteActionLabel, icon: <Delete /> },
+        {
+          id: "duplicate",
+          label: labels?.duplicateActionLabel,
+          icon: <Duplicate />,
+        },
+      ],
+    [labels?.deleteActionLabel, labels?.duplicateActionLabel, nodeActionsProp],
+  );
+
+  useMemo(() => {
+    registerNode(id, {
+      label: title || "",
+      inputs,
+      outputs,
+    });
+    return () => unregisterNode(id);
+  }, [id, title, inputs, outputs, registerNode, unregisterNode]);
+
+  const inputEdges = useFlowNodeInputEdges();
+  const outputEdges = useFlowNodeOutputEdges();
+
+  const handleDefaultAction = useCallback(
+    (action: HvFlowNodeAction) => {
+      if (!node) return;
+
+      if (action.callback) {
+        action.callback(node);
+        return;
+      }
+
+      // built-in actions
+      switch (action.id) {
+        case "delete":
+          reactFlowInstance.deleteElements({ nodes: [node] });
+          break;
+        case "duplicate":
+          reactFlowInstance.addNodes([
+            {
+              ...node,
+              id: uid(),
+              position: {
+                x: node.position.x,
+                y: node.position.y + (node.height || 0) + 20,
+              },
+              selected: false,
+              zIndex: Number(theme.zIndices.overlay),
+            },
+          ]);
+          break;
+        default:
+          break;
+      }
+    },
+    [node, reactFlowInstance],
+  );
+
+  return {
```

Comment:
I think the following props shouldn't be exposed to the user:
- `registerNode`
- `unregisterNode`